core/src/main/java/com/uber/m3/tally/ScopeImpl.java

```diff
@@ -280,15 +281,25 @@ private Scope subScopeHelper(String prefix, Map<String, String> tags) {
 
         String key = keyForPrefixedStringMap(prefix, mergedTags);
 
+        return computeSubscopeIfAbsent(key, mergedTags);
+    }
+
+    // This method must only be called on unit tests or benchmarks
```

Comment:
nit: this comment is confusing as this function is in fact called in the method above. I think the fact that you made it protected is already good enough indicator that it should not be called outside of this package

Reply:
I'm doing this because we don't have an annotation to ensure that it is only called this way. Something like https://developer.android.com/reference/androidx/annotation/VisibleForTesting. Just wanted to be more explicit here.